homeassistant/components/shelly/coordinator.py

```diff
@@ -536,6 +543,8 @@ async def _async_update_data(self) -> None:
             raise UpdateFailed(f"Device disconnected: {repr(err)}") from err
         except InvalidAuthError:
             self.entry.async_start_reauth(self.hass)
+        except FirmwareUnsupported as err:
+            raise ConfigEntryNotReady(repr(err)) from err
```

Comment:
```suggestion
            raise ConfigEntryNotReady from err
```

https://github.com/home-assistant/core/pull/109076/files#r1470309887